### PR TITLE
Add support for cuDNN to BatchNormalization and LocalResponseNormalization

### PR DESCRIPTION
**Pull request by saudet**
Also adjust a couple of things with CudnnConvolutionHelper and CudnnSubsamplingHelper. Before merging this in though, please let me know if it looks alright. And with that, DL4J will support more of cuDNN than TensorFlow. Yay! :)

### CODE REVIEW

**Comment by agibsonccc** [comment]
LGTM
